test/powershell/Modules/Microsoft.PowerShell.Utility/Json.Tests.ps1

```diff
@@ -345,6 +345,258 @@ Describe "Json Tests" -Tags "Feature" {
             ValidateSampleObject -result $result -hasEmbeddedSampleObject
         }
 
+		It "ConvertFrom-Json correctly processes datetimes vs strings using newtonsoft conventions" {
+			# Create all the various "standard" date formats from a source date 2008-09-22T14:01:54.9571247Z
+			# Then we deserialize them and verify which ones newtonsoft decides to make into datetimes and which ones newtonsoft keeps as strings
+			# For the ones that are datetimes, make sure they retain all of the various formats correctly
+			# for the ones that are strings just verify that they are actually strings.
+
+			$json = @"
+{
+	"date-s-should-parse-as-datetime": "2008-09-22T14:01:54",
+	"date-upperO-should-parse-as-datetime": "2008-09-22T14:01:54.9571247Z",
+	
+	"date-o-should-parse-as-string": "2019-12-17T06:14:06 +06:00",
+	"date-upperD-should-parse-as-string": "Monday, September 22, 2008",
```

Comment:
The tests are culture sensitive. I think it is too difficult to make it culture insensitive.
I suggest to skip the test if not en-US but we should check what is culture on our CI Linux (I guess here it is invariant culture) and CI MacOS.
```powershell
            if ((Get-Culture).Name -neq "en-US") {
                Set-ItResult -Skipped -Because "Skip the test because culture is not en-US."
            }
```

Reply:
Did you see a failure?

I will try to see if I can change my system culture and get these to fail but the string tests are just a string comparison and should be culture-insensitive. Newtonsoft keeps them as strings so we're just doing a string equals.

Newtonsoft only converts ISO 8601 dates by default which would be as formatted `2008-09-22T14:01:54` and `2008-09-22T14:01:54.9571247Z` which I think should also be culture-insensitive?

Reply:
> "Monday, September 22, 2008"

This depends on locale.
And we have explicit checks like:
`$result."date-upperD-should-parse-as-string" | Should -Be "Monday, September 22, 2008"`

Reply:
So I am not trying to be difficult -- sorry if I'm being silly.

My understanding is that this is just taking a string, deserializing it as a string and making sure it's the same string. Basically I'm saying:

```
$json = '{"value": "Monday, September 22, 2008"}'
$string = convertfrom-json $json
$string.value -is [string]
$string.value -eq $true
```

As this never becomes a datetime the locale shouldn't have an impact. So I *think* these are fine

You're totally right about the actual datetime checks that are below these though but it looks pretty easy to specify the culture as `[datetime]` `ToString()` can take a format and a cultureinfo:

```
PS C:\Users\user> ([datetime]::utcnow).tostring("D", [cultureinfo]::GetCultureInfo('sv-SE'))
torsdag 15 juli 2021
PS C:\Users\user> ([datetime]::utcnow).tostring("D", [cultureinfo]::GetCultureInfo('en-us'))
Thursday, July 15, 2021

Would you be alright with me changing these to hardcode the en-us culture instead of conditionally disabling the tests?

Reply:
> Would you be alright with me changing these to hardcode the en-us culture instead of conditionally disabling the tests? 

Yes.